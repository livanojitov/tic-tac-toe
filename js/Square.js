class Square extends React.Component{

  render(){
    const { win, disableSquare, handleClick, id, player } = this.props;

    return (
      <button style={{ backgroundImage : `url(images/${player}.jpg)`}}
              className = {win}
              disabled  = {disableSquare} 
              onClick   = {(e) => { if (handleClick){ handleClick(e)} }}
              id        = {id}>
      </button>
    )
  }
}