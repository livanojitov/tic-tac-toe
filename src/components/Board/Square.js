import React from "react";
import PropTypes from "prop-types";
import "./Board.scss";

const Square = React.memo(({ disabled, id, image, win }) => {
  const firebaseUrl = `url(https://firebasestorage.googleapis.com/v0/b/puzzle-ebd10.appspot.com/o/images%2F${image}.jpg?alt=media)`;

  return (
    <button
      className={win}
      data-testid="square"
      disabled={disabled}
      id={id}
      style={{ backgroundImage: firebaseUrl }}
    ></button>
  );
});

Square.propTypes = {
  disabled: PropTypes.bool,
  id: PropTypes.number,
  image: PropTypes.string,
  win: PropTypes.string,
};

export default Square;
