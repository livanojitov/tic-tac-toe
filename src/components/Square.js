import React from 'react';

const Square = ({ win, disabled, handleClick, id, player }) => {
  return (
    <button style     = {{ backgroundImage : `url(images/${player}.jpg)`}}
            className = {win}
            disabled  = {disabled} 
            onClick   = {(e) => { if (handleClick){ handleClick(e)} }}
            id        = {id}>
    </button>
  )
}

export default Square;