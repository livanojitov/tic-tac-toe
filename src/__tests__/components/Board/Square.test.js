import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Square from "../../../components/Board/Square";

describe("Square", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Square />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should apply the props", () => {
    const url =
      "https://firebasestorage.googleapis.com/v0/b/puzzle-ebd10.appspot.com/o/images%2Fdefault.jpg?alt=media";

    render(<Square win="win" disabled={true} id={100} image={"default"} />);

    expect(screen.getByTestId("square")).toHaveAttribute("class", "win");
    expect(screen.getByTestId("square")).toHaveAttribute("disabled");
    expect(screen.getByTestId("square")).toHaveAttribute("id", "100");
    expect(screen.getByTestId("square")).toHaveAttribute(
      "style",
      expect.stringContaining(url)
    );
  });
});
