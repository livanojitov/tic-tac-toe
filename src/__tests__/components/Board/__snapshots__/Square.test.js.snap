// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square should render correctly 1`] = `
<DocumentFragment>
  <button
    data-testid="square"
    style="background-image: url(https://firebasestorage.googleapis.com/v0/b/puzzle-ebd10.appspot.com/o/images%2Fundefined.jpg?alt=media);"
  />
</DocumentFragment>
`;
